src/include/ipc/topology.h

```diff
@@ -46,6 +46,7 @@ enum sof_comp_type {
 	SOF_COMP_DEMUX,
 	SOF_COMP_ASRC,		/**< Asynchronous sample rate converter */
 	SOF_COMP_DCBLOCK,
+	SOF_COMP_SMART_AMP,		/**< smart amplifier component */
```

Comment:
Component types are going away. Lets not add any new ones.

Reply:
I just had the same remark on the kernel side, even if we kept these comp_type it should be identified as 3rd-party stuff

Reply:
@plbossart as replied on the kernel side, we plan to add config item for the 3rd-party algorithm specific selection, to the component itself, we want to make it a generic one with dummy(simple enough) algorithm, so if someone contribute an better open source algorithm, we can create an item for it also (something like a N->1 mux selection in the config menu).